jessetilro/tud-gm-1
# Registration

# Changes

Implemented the complete algorithm for registration functionality. Updated compilation and run scripts. Added registration problem projects and configured one to be launched by default.

# How to test

1. Invoke compilation and run script, which will automatically launch the registration problem project
2. Go to Method > Our Thingy > Surface Registrator Deluxe
3. Select for P the partial geometry, and for Q the full head
4. Press the button and monitor the console for activity
5. When finished, press `Method > Mark > Unmark All Vertices` to see result (somehow not effectuated directly, at least not with me)
6. *Optionally* tweak the `n` and `iterations` class variables in the `Registration` class, and start over from step 1

# How to improve

- Make sure the transformations performed on the geometry as part of the registration process are flushed and made visible directly
- Find out why registration does not yield desired result yet and fix it
- Add option to use point to plane distance
- Add options to set sample size (n) and number of iterations in the interface panel

# How to review

Just merge this after you both reviewed it. I won't be able to make any changes anymore, and this branch is stable. So if you like to improve this, I would suggest to merge this, branch from master and make the changes a new PR.
